### ISSUE TITLE
lolcat --help segfaults

### ISSUE THREAD

**Issue by LunarLambda** [jaseg/lolcat#15]
Freshly compiled on Ubuntu 18.04 with the latest make and gcc packages.
![image](https://user-images.githubusercontent.com/38919842/43984291-eb9fe3aa-9cff-11e8-93ba-d50053cd1663.png)

**Comment by LunarLambda** [comment]
```
lunar-base% lolcat --help
zsh: segmentation fault (core dumped)  lolcat --help
lunar-base% strace lolcat --help
execve("/usr/local/bin/lolcat", ["lolcat", "--help"], [/* 13 vars */]) = 0
brk(NULL)                               = 0x8a7000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=44345, ...}) = 0
mmap(NULL, 44345, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f6d9cbab000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/x86_64-linux-gnu/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0P\t\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1868984, ...}) = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6d9cba0000
mmap(NULL, 3971488, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f6d9c430000
mprotect(0x7f6d9c5f0000, 2097152, PROT_NONE) = 0
mmap(0x7f6d9c7f0000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1c0000) = 0x7f6d9c7f0000
mmap(0x7f6d9c7f6000, 14752, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f6d9c7f6000
close(3)                                = 0
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6d9cb90000
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f6d9cb80000
arch_prctl(ARCH_SET_FS, 0x7f6d9cb90700) = 0
mprotect(0x7f6d9c7f0000, 16384, PROT_READ) = 0
mprotect(0x601000, 4096, PROT_READ)     = 0
mprotect(0x7f6d9ca25000, 4096, PROT_READ) = 0
munmap(0x7f6d9cbab000, 44345)           = 0
gettimeofday({1535135114, 229292}, NULL) = 0
brk(NULL)                               = 0x8a7000
brk(0x8c8000)                           = 0x8c8000
open("/usr/lib/locale/locale-archive", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=1668976, ...}) = 0
mmap(NULL, 1668976, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f6d9c88d000
close(3)                                = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_ACCERR, si_addr=0x7fffc8b89fa0} ---
+++ killed by SIGSEGV (core dumped) +++
zsh: segmentation fault (core dumped)  strace lolcat --help
```

**Comment by XenGi** [comment]
mhk. Then maybe @jaseg can help?

**Comment by jaseg** [comment]
@LunarLambda are you absolutely, 100% certain that the lolcat that you're running and that's segfaulting is the one you built from the current master branch? I'm asking since I recently fixed the glibc segfault issue by removing the problematic call to ```fmemopen``` altogether, and according to your strace your segfaulting lolcat binary calls mmap, which it really shouldn't since it's not using ```fmemopen``` anymore.

**Comment by jaseg** [comment]
I'm closing this for now since I believe this issue is fixed in v1.0. If you think it's not, please just reopen this.
